include/ua_plugin_ca.h

```diff
@@ -0,0 +1,91 @@
+/* This work is licensed under a Creative Commons CCZero 1.0 Universal License.
+ * See http://creativecommons.org/publicdomain/zero/1.0/ for more information. 
+ *
+ *    Copyright 2018 (c) Markus Karch, Fraunhofer IOSB
+ */
+
+#ifndef OPEN62541_UA_PLUGIN_CA_H
+#define OPEN62541_UA_PLUGIN_CA_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "ua_types.h"
+#include "ua_plugin_log.h"
+#include "ua_types_generated.h"
+
+#ifdef UA_ENABLE_GDS_CM
+
+/*
+ * A GDSCertificateGroup represents an interface to one CA.
+ * Currently only one plugin interface is implemented using GnuTLS.
+ * */
+struct UA_GDS_CA;
+typedef struct UA_GDS_CA UA_GDS_CA;
+
+struct UA_GDS_CA {
```

Comment:
Can you check which parameters for all the functions below can be marked as `const`?

In general we should try to mark as much as possible as const.
https://wiki.sei.cmu.edu/confluence/display/c/DCL13-C.+Declare+function+parameters+that+are+pointers+to+values+not+changed+by+the+function+as+const

E.g.:
```
    UA_StatusCode (*createNewKeyPair) (UA_GDS_CA *scg,
                                       UA_String *subjectName,
                                       UA_String *privateKeyFormat,
                                       UA_String *privateKeyPassword,
                                       unsigned  int keySize,
                                       size_t domainNamesSize,
                                       UA_String *domainNamesArray,
                                       UA_ByteString *certificate,
                                       UA_ByteString *privateKey,
                                       size_t *issuerCertificateSize,
                                       UA_ByteString **issuerCertificates);

```

could be changed to:

```
    UA_StatusCode (*createNewKeyPair) (UA_GDS_CA *scg,
                                       const UA_String *subjectName,
                                       const UA_String *privateKeyFormat,
                                       const UA_String *privateKeyPassword,
                                       unsigned  int keySize,
                                       size_t domainNamesSize,
                                       const UA_String *domainNamesArray,
                                       UA_ByteString *certificate,
                                       UA_ByteString *privateKey,
                                       size_t *issuerCertificateSize,
                                       UA_ByteString **issuerCertificates);

```

Reply:
See also:
http://gamesfromwithin.com/the-const-nazi

And funny to watch:
http://gamesfromwithin.com/wp-content/uploads/2010/07/const_nazi.mov